services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedTenantService.java

```diff
@@ -0,0 +1,630 @@
+                .compose(successExistingCollections -> {
+                    if (successExistingCollections.contains(getConfig().getCollectionName())) {
+                        return Future.succeededFuture();
+                    } else {
+                        // create index & implicit collection
+                        return mongoDbCallExecutor.createCollectionIndex(getConfig().getCollectionName(),
+                                new JsonObject().put(TenantConstants.FIELD_PAYLOAD_TENANT_ID, 1).put(TenantConstants.FIELD_PAYLOAD_DEVICE_ID, 1),
+                                new IndexOptions().unique(true));
+                    }
+                })
+                .compose(success -> {
+                    if (getConfig().isModificationEnabled()) {
+                        log.info("persistence is disabled, will not save tenant identities to mongoDB.");
+                    }
+                    log.debug("startup complete");
+                    startPromise.complete();
+                    return Future.succeededFuture();
+                }).onFailure(reason -> {
+            log.error("Index creation failed", reason);
+            startPromise.fail(reason.toString());
+        });
+    }
+
+    /**
+     * Stops the service.
+     *
+     * @param stopPromise the shutdown promise
+     */
+    @Override
+    public void stop(final Promise<Void> stopPromise) {
+        this.mongoClient.close();
+        stopPromise.complete();
+    }
+
+    /**
+     * Updates data of a present tenant.
+     * <p>
+     * Tenant modification have to be enabled in {@link MongoDbBasedTenantsConfigProperties} or exits with {@code HttpURLConnection.HTTP_FORBIDDEN}.
+     * </p>
+     * <p>
+     * Provided version {@code resourceVersion} have to be empty or equal to the version of present tenant.
+     * A provided <em>certificate authority</em> must not be set in other tenants.
+     * </p>
+     *
+     * @param tenantId        The identifier of the tenant.
+     * @param tenantObj       The updated configuration information for the tenant (may be {@code null}).
+     * @param resourceVersion The identifier of the resource version to update.
+     * @param span            The active OpenTracing span for this operation. It is not to be closed in this method!
+     *                        An implementation should log (error) events on this span and it may set tags and use this span as the
+     *                        parent for any spans created in this method.
+     * @return Result of update operation: On success {@code HttpURLConnection.HTTP_NO_CONTENT}, http error codes else. {@code HttpURLConnection.HTTP_FORBIDDEN} if modification is disabled.
+     * @throws NullPointerException if {@code tenantId} or {@code tenantObj} are {@code null}.
+     */
+    @Override
+    public Future<OperationResult<Void>> updateTenant(final String tenantId, final Tenant tenantObj, final Optional<String> resourceVersion, final Span span) {
```

Comment:
You could just remove this _javadoc_ for this overridden method, as the interface `TenantManagementService` has a well documented one. If you still prefer to add one, you could any additional information that you have regarding this overridden method. If you add one, please follow the same style as of the `TenantManagementService`. Also please check the _javadoc_ of the other methods too.